src/Gaufrette/Adapter/OpenStack.php

```diff
@@ -0,0 +1,377 @@
+<?php
+
+namespace Gaufrette\Adapter;
+
+use Gaufrette\Adapter;
+use Gaufrette\Exception\FileNotFound;
+use Gaufrette\Exception\StorageFailure;
+use Gaufrette\Util;
+use OpenStack\Common\Error\BadResponseError;
+use OpenStack\ObjectStore\v1\Models\Container;
+use OpenStack\ObjectStore\v1\Models\StorageObject;
+use OpenStack\ObjectStore\v1\Service;
+
+/**
+ * OpenStack adapter.
+ *
+ * @author  James Watson <james@sitepulse.org>
+ * @author  Daniel Richter <nexyz9@gmail.com>
+ * @author  Nicolas MURE <nicolas.mure@knplabs.com>
+ *
+ * @see http://docs.os.php-opencloud.com/en/latest/services/object-store/v1/objects.html
+ * @see http://refdocs.os.php-opencloud.com/OpenStack/OpenStack.html
+ */
+final class OpenStack implements Adapter,
+                           ChecksumCalculator,
+                           ListKeysAware,
+                           MetadataSupporter,
+                           MimeTypeProvider,
+                           SizeCalculator
+{
+    /**
+     * @var Service
+     */
+    private $objectStore;
+
+    /**
+     * @var string
+     */
+    private $containerName;
+
+    /**
+     * @var bool
+     */
+    private $createContainer;
+
+    /**
+     * @var Container
+     */
+    private $container;
+
+    /**
+     * @param Service $objectStore
+     * @param string  $containerName   The name of the container
+     * @param bool    $createContainer Whether to create the container if it does not exist
+     */
+    public function __construct(Service $objectStore, $containerName, $createContainer = false)
+    {
+        $this->objectStore = $objectStore;
+        $this->containerName = $containerName;
+        $this->createContainer = $createContainer;
+    }
+
+    /**
+     * Returns an initialized container.
+     *
+     * @throws StorageFailure
+     *
+     * @return Container
+     */
+    private function getContainer()
+    {
+        if ($this->container) {
+            return $this->container;
+        }
+
+        try {
+            if ($this->objectStore->containerExists($this->containerName)) {
+                return $this->container = $this->objectStore->getContainer($this->containerName);
+            }
+
+            if (!$this->createContainer) {
+                throw new StorageFailure(sprintf('Container "%s" does not exist.', $this->containerName));
+            }
+
+            return $this->container = $this->createContainer();
```

Comment:
I'm not sure about this one too, I guess there're some security-related configuration that might be done on OpenStack storage containers :thinking:

Reply:
So better drop the auto creation of containers and let the user do it on its side ?
(ie drop the `createContainer` flag from the adapters constructor)
Seems reasonable indeed. Let me know if you agree and I'll update the roadmap for v1.

Reply:
iirc we already discussed that but for the record: yeah, this should be removed from all the adapters.

Reply:
Yes :+1: it has been added to the v1 roadmap (#618)